File: app/src/main/java/com/xuvjso/nfmovies/UI/PopupMenu.java
package com.xuvjso.nfmovies.UI;

import android.content.Context;
import android.view.Gravity;
import android.view.View;
import android.view.WindowManager;
import android.widget.*;
import android.widget.PopupWindow;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.xuvjso.nfmovies.Adapter.PopupMenuAdapter;
import com.xuvjso.nfmovies.Listener.PopupMenuItemClickListener;
import com.xuvjso.nfmovies.R;

import java.util.List;

public class PopupMenu extends PopupWindow {
    private ImageButton closeButton;
    private RecyclerView menuRv;
    private PopupMenuItemClickListener clickListener;
    private Context context;

    public PopupMenu(Context context, View contentView, List<String> items, PopupMenuItemClickListener clickListener) {
        super(contentView, WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.MATCH_PARENT, true);
        closeButton = contentView.findViewById(R.id.menu_close_button);
        menuRv = contentView.findViewById(R.id.menu_recyclerview);
        closeButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dismiss();
            }
        });
        PopupMenuAdapter adapter = new PopupMenuAdapter(context, items, clickListener);
        menuRv.setAdapter(adapter);
        menuRv.setLayoutManager(new LinearLayoutManager(context, RecyclerView.VERTICAL, false));
    }

    public void show(View parent) {
        showAtLocation(parent, Gravity.TOP | Gravity.RIGHT, 0,0);
    }

}

File: app/src/main/java/com/futuretech/nfmovies/UI/PopupMenu.kt
package com.futuretech.nfmovies.UI

import android.content.Context
import android.view.Gravity
import android.view.View
import android.view.WindowManager
import android.view.animation.AnimationUtils
import android.widget.*
import android.widget.PopupWindow
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.futuretech.nfmovies.Adapter.PopupMenuAdapter
import com.futuretech.nfmovies.Listener.PopupMenuItemClickListener
import com.futuretech.nfmovies.R

class PopupMenu(context: Context, contentView: View, items: List<String>, clickListener: PopupMenuItemClickListener) : PopupWindow(contentView, WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.MATCH_PARENT, true) {
    private val closeButton: ImageButton
    private val menuRv: RecyclerView
    private val clickListener: PopupMenuItemClickListener? = null
    private val context: Context? = null

    init {
        closeButton = contentView.findViewById(R.id.menu_close_button)
        menuRv = contentView.findViewById(R.id.menu_recyclerview)
        closeButton.setOnClickListener { dismiss() }
        val adapter = PopupMenuAdapter(context, items, clickListener)
        menuRv.adapter = adapter
        menuRv.layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
        animationStyle = R.style.popup_animation
    }

    fun show(parent: View) {
        showAtLocation(parent, Gravity.TOP or Gravity.END, 0, 0)
    }

}
